tpe/lib/src/CollisionDetector.cc

```diff
@@ -31,14 +31,26 @@ std::vector<Contact> CollisionDetector::CheckCollisions(
   // contacts to be filled and returned
   std::vector<Contact> contacts;
 
+  // cache of collide bitmasks
+  std::map<std::size_t, uint16_t> collideBitmasks;
+
```

Comment:
A few comments about this.
First of all you can probably see coming my "usual" feedback about using hash maps, since we are also doing quite a lot of operations on this map (number of models in the simulation squared).
But I am also not too sure about using a cache in the first place, from what I can see we are trying to avoid the overhead of iterating over the children in the `Entity::GetCollideBitmask()` function, what do you think about calculating / caching the entity bitmask in the Entity object in the first place?

Also small comment in the worldAabb map, if it's not out of the scope of the PR I would still suggest changing to a hash map to save a logn binary search in every find() and insert operation. All the function calls should be unchanged since the API is quite similar.

Reply:
ok yeah I think the cache for collide bitmask is probably premature optimization so I just removed it. The `GetCollideBitmask` function is just doing bitwise OR operations which should be fast. Updated `worldAabb` to use `unordered_map`. 1555caa

Reply:
I'm still quite interested in the idea of caching the collide bitmask at the entity level (i.e. whenever a "child" is added / removed recalculate the bitmask and make the Get function be a simple property getter), do you think that would be a good approach? Otherwise this looks good to me!

Reply:
I tested the caching approach and ended up adding more code than I had planned so I created a separate pull request for it, #73. Feedback welcome.

Reply:
ok merged #73 into this PR